test_common/harness/deviceInfo.cpp

```diff
@@ -13,13 +13,18 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+
 #include <stdexcept>
+#include <vector>
+
 #include "deviceInfo.h"
 #include "errorHelpers.h"
 #include "typeWrappers.h"
 
-/* Helper to allocate and return a buffer containing device information for the specified device info parameter. */
-static void *alloc_and_get_device_info( cl_device_id device, cl_device_info param_name, const char *param_description )
+/* Helper to return a string containing device information for the specified
+ * device info parameter. */
+static std::string get_device_info(cl_device_id device,
+                                   cl_device_info param_name)
```

Comment:
This signature only really works when the returned value is a string. This works for all current uses so I'm inclined to leave it as is.

Reply:
Renamed to `get_device_info_string` to make this more obvious.